src/core/utils/url.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useEffect, useState } from "react";
+import { RootState, store, useSelector } from "../store";
+import { addUrlEntries } from "../url.slice";
+
+export const useUrls = () => {
+  const { data } = useSelector((state: RootState) => state.url);
+  return data;
+};
+
+export const withUrl = <T,>(Component: React.ComponentType<T>) => {
```

Comment:
Consider renaming this to `withUrls` as there are multiple and our hook is called `useUrls`.

Reply:
donno